optimum/onnxruntime/modeling_ort.py

```diff
@@ -569,6 +577,64 @@ def from_pretrained(
             **kwargs,
         )
 
+    def prepare_output_buffer(self, output_shape, output_name: str):
+        """Prepares the buffer of output_name with a 1D tensor."""
+        ort_type = TypeHelper.get_output_type(self.model, output_name)
+        torch_type = TypeHelper.ort_type_to_torch_type(ort_type)
+        output_buffer = torch.empty(np.prod(output_shape), dtype=torch_type, device=self.device).contiguous()
+        return output_buffer
+
+    def prepare_io_binding(self, *model_inputs):
+        io_binding = self.model.io_binding()
+
+        input_name_to_tensor = {}
+        for idx, tensor in enumerate(model_inputs):
+            if tensor is None:
+                continue
+            if idx >= len(self.model_input_names):
+                raise ValueError(
+                    "Too many inputs were provided here, the underlying ONNX model has the following inputs: "
+                    f"{', '.join(self.model_input_names)}"
+                )
+            name = self.model_input_names[idx]
+            input_name_to_tensor[name] = tensor
+            tensor = tensor.contiguous()
+            io_binding.bind_input(
+                name,
+                tensor.device.type,
+                self.device.index,
+                self.name_to_np_type[name],
+                tuple(tensor.shape),
+                tensor.data_ptr(),
+            )
+
+        dimensions = {}
+        for input_ in self.model.get_inputs():
+            shape = input_.shape
+            for idx, axis in enumerate(shape):
+                if isinstance(axis, str):
+                    dimensions[axis] = input_name_to_tensor[input_.name].shape[idx]
+
+        output_shapes = {}
+        output_buffers = {}
+        for output_node in self.model.get_outputs():
+            output_name = output_node.name
+            output_shape = tuple(map(lambda x: dimensions.get(x, x), output_node.shape))
```

Comment:
How are you going to deal with shapes on dynamic axis? The definition could change according to task, got to be careful.

Reply:
We are getting them when possible, otherwise we can specify them via the `known_output_shapes` argument.